utils/s3/client.hh

```diff
@@ -32,11 +32,17 @@ class client : public enable_shared_from_this<client> {
 
 private:
     future<> get_object_header(sstring object_name, http::experimental::client::reply_handler handler);
+    std::time_t parse_http_last_modified_time(const sstring& object_name, sstring last_modified);
```

Comment:
nit: It doesn't have to be client member at all. As you noticed in the comment, it even deserves sitting in http client code
  dependent-nit: if member -- can be marked const
nit2: noexcept in any case :)